### ISSUE TITLE
Styles.xml sometimes gets corrupted

### ISSUE THREAD

**Issue by Hilding76** [PHPOffice/PHPExcel#508]
Hi 
I open templates, write and save, in two modes: batched or manually
Some reports when writing in the batchmode gets corrupted styles.xml
This seem to depend on the reports and the order of reports generated in that batch.
Some parts of my code:
opening templates
if(substr($inputfile[$i-1], strlen($inputfile[$i-1])-4,4)==".xls")
						{
							if (file_exists($inputfile[$i-1]."x"))
							{
								try
								{
									$objPHPExcel=PHPExcel_IOFactory::load($inputfile[$i-1]."x");
								}
								catch( Exception $e ) 
								{
									echo 'Problem att öppna mall '.$inputfile[$i-1]."x".'<br>';
									echo $e->getMEssage();
								}	
								$templatename=$inputfile[$i-1]."x";
								// nedan är workaround mot att phpexcel skapar extraflikar för xlsx..
								$reader = new PHPExcel_Reader_Excel2007();
								$working_sheet_name_list =$reader->listWorksheetNames($inputfile[$i-1]."x");
								$savesheetnr=$objPHPExcel->getActiveSheetIndex();
								$worksheetworkaround= false;
								$stepadjust=0;
								foreach($objPHPExcel->getSheetNames() as $index=>$name)
								{
								 foreach($working_sheet_name_list as $keep)
								 {
									if ($name==$keep)
										$worksheetworkaround= true;
								 }
								 if ($worksheetworkaround== false)
								 { // radera...
									$index=$index-$stepadjust;
									$objPHPExcel->removeSheetByIndex($index);
									$stepadjust++;
								 }
								 $worksheetworkaround= false;
								}
							}
							else
							{
								try
								{
									$objPHPExcel=PHPExcel_IOFactory::load($inputfile[$i-1]);
								}
								catch( Exception $e ) 
								{
									echo 'Problem att öppna mall '.$inputfile[$i-1].'<br>';
									echo $e->getMEssage();
								}	
								$templatename=$inputfile[$i-1];
								$savesheetnr=$objPHPExcel->getActiveSheetIndex();
								$reader = new PHPExcel_Reader_Excel2007();
								$working_sheet_name_list =$reader->listWorksheetNames($inputfile[$i-1]);
								$worksheetworkaround= false;
								$stepadjust=0;
								foreach($objPHPExcel->getSheetNames() as $index=>$name)
								{
								 foreach($working_sheet_name_list as $keep)
								 {
									if ($name==$keep)
										$worksheetworkaround= true;
								 }
								 if ($worksheetworkaround== false)
								 { // radera...
									$index=$index-$stepadjust;
									$objPHPExcel->removeSheetByIndex($index);
									$stepadjust++;
								 }
								 $worksheetworkaround= false;
								}
							}
						}
						else
						{
							try
							{
								$objPHPExcel=PHPExcel_IOFactory::load($inputfile[$i-1]);
							}
							catch( Exception $e ) 
							{
								echo 'Problem att öppna mall '.$inputfile[$i-1].'<br>';
								echo $e->getMEssage();
							}	
							$templatename=$inputfile[$i-1];
							$savesheetnr=$objPHPExcel->getActiveSheetIndex();
							$reader = new PHPExcel_Reader_Excel2007();
							$working_sheet_name_list =$reader->listWorksheetNames($inputfile[$i-1]);
							$worksheetworkaround= false;
							$stepadjust=0;
							foreach($objPHPExcel->getSheetNames() as $index=>$name)
							{
							 foreach($working_sheet_name_list as $keep)
							 {
								if ($name==$keep)
									$worksheetworkaround= true;
							 }
							 if ($worksheetworkaround== false)
							 { // radera...
								$index=$index-$stepadjust;
								$objPHPExcel->removeSheetByIndex($index);
								$stepadjust++;
							 }
							 $worksheetworkaround= false;
							}
						}

Writing values
if (is_numeric($fieldvalue))
									{
										$nummeric=true;
										$numertype=PHPExcel_Cell_DataType::TYPE_NUMERIC;
									}
                                                                              if($colwrite)
										{
											if (is_null($fieldvalue) == false and $fieldvalue!= "")
											{
												if($nummeric)
												{
													$objPHPExcel->getActiveSheet()->setCellValueExplicitByColumnAndRow($startx+$j-1,$starty+$k,$fieldvalue,$numertype);
												}
												else
												{
												if ($setcelltype)
												{

													$objPHPExcel->getActiveSheet()->setCellValueExplicitByColumnAndRow($startx+$j-1,$starty+$k,$fieldvalue,$translate_types[gettype($fieldvalue)]);

												}
												else
												{
												 $objPHPExcel->getActiveSheet()->setCellValueByColumnAndRow($startx+$j-1,$starty+$k,$fieldvalue);
												}
												}
											}
											
											if($cutting)
											{
												if (is_null($fieldvalue)  or $fieldvalue== "")
												{
													$objPHPExcel->getActiveSheet()->removeRow($starty+$k,1);
													$k--;
												}
											}
											if($skip_if_null)
											{
												if (is_null($fieldvalue)  or $fieldvalue== "")
												{
													$k--;
												}
											}		
										}
										else
										{
											if (is_null($fieldvalue) == false and $fieldvalue!= "")
											{
												if($nummeric)
												{
													$objPHPExcel->getActiveSheet()->setCellValueExplicitByColumnAndRow($startx+$k-1,$starty+$j,$fieldvalue,$numertype);
												}
												else
												{
											
													if ($setcelltype)
													{		
														$objPHPExcel->getActiveSheet()->setCellValueExplicitByColumnAndRow($startx+$k-1,$starty+$j,$fieldvalue,$translate_types[gettype($fieldvalue)]);
													}
													else
													{	
														$objPHPExcel->getActiveSheet()->setCellValueByColumnAndRow($startx+$k-1,$starty+$j,$fieldvalue);
													}
												}
											}
											if($cutting)
											{
												if (is_null($fieldvalue)  or $fieldvalue== "")
												{
													$objPHPExcel->getActiveSheet()->removeColumn($startx+$k-1,1);
													$k--;
												}
											}
											if($skip_if_null)
											{
												if (is_null($fieldvalue)  or $fieldvalue== "")
												{
													//$objPHPExcel->getActiveSheet()->removeColumn($startx+$k-1,1);
													$k--;
												}
											}
										}
								}

Saving files
$objPHPExcel->setActiveSheetIndex($savesheetnr);
$resulting_filename=save_file($pathtowrite,$filetowrite,$objPHPExcel,$filetype);

							if ($onthefly==true and $debug==false)
						{
								if (array_key_exists("Save_as_exported",$args))
								{
									write_metadatafile($pathtowrite, $resulting_filename,$metadata_type, $norows, $premisions);
								}

								set_time_limit(0);
								ob_clean();
								 if(ini_get('zlib.output_compression'))
										ini_set('zlib.output_compression', 'Off');
								header("Expires: Mon, 26 Jul 1997 05:00:00 GMT");
								header("Last-Modified: " . gmdate("D, d M Y H:i:s") . " GMT");
								header("Cache-Control: no-store, no-cache, must-revalidate");
								header("Cache-Control: post-check=0, pre-check=0", false);
								header("Pragma: no-cache");
								ob_start(); 
								$fHandler = new FileFuncs();
								if (array_key_exists("Save_as_exported",$args))
								{
									$fHandler->FileDownload($pathtowrite.$resulting_filename, false, true);
								}
								else
								{
									$fHandler->FileDownload($pathtowrite.$resulting_filename, true, true);
								}
								ob_end_flush();	 
						}
						else
						{
							$templatepath=substr($templatename,0,strrpos($templatename,'/')+1);
							$templatefilename=substr($templatename,strrpos($templatename,'/')+1,strlen($templatename)-strrpos($templatename,'/')-1);
							if(substr($templatefilename, strlen($templatefilename)-5,5)==".xlsx" and $cutfile==false)
							{
								fixformat($pathtowrite, $resulting_filename,$templatepath,$templatefilename);
							}
							log_exported_file($pathtowrite, $resulting_filename,$sql,$conn);
							write_metadatafile($pathtowrite, $resulting_filename,$metadata_type, $norows, $premisions);
							$cutfile=false;
						}
}
					  catch( Exception $e ) {
							echo 'Problem att spara '.$filetowrite.'<br>';
							echo $e->getMEssage();
					  }

function save_file($path,$savename,$objPHPExcel,$filetype)
{
	$filer=2;
	$paths=explode("\\",$path);
	
	$path2='';
	foreach($paths as $path3)
	{
		if($path3<> '')
		{
			if ($path3[1]==":")
			{
				$path2=$path3."\\";
			}
			elseif($path3<> '')
			{
					$path2=$path2.$path3."\\";
					if (file_exists($path2)==false)
					{
						mkdir($path2);
						
					}
			}
		}
	}
	if (file_exists($path.$savename))
	{
	 $savename=strrev(substr(strrev($savename),strpos(strrev($savename),'.')+1)).'_'.$filer.'.'.strrev(substr(strrev($savename),0,strpos(strrev($savename),'.')));

	 while (file_exists($path.$savename))
	 {
		$filer++;
		$savename=strrev(substr(strrev($savename),strpos(strrev($savename),'_')+1)).'_'.$filer.'.'.strrev(substr(strrev($savename),0,strpos(strrev($savename),'.')));
	 }
	 
	}
	$justthefilename=$savename;
	$savename=$path.$savename;
		switch($filetype)
		{
			case 'excel':
				$otherfiletype= 'Excel2007';//'Excel5';
				break;
			case 'excel2007':
				$otherfiletype='Excel2007';
				break;
			case 'xlsx':
				$otherfiletype='Excel2007';
				break;				
			case 'csv':
				$otherfiletype='csv';
				break;
			case 'pdf':
				$otherfiletype='PDF';
				break;
			default:
				throw new Exception("Tyvärr så är inte ".$filetype." definerad (än)");
		
		}
	

	
		try
		{
			$objWriter = PHPExcel_IOFactory::createWriter($objPHPExcel, $otherfiletype);
			if ($otherfiletype=='PDF')
			{
				$objWriter->writeAllSheets();
			}
		}
		catch( Exception $e ) 
		{
				echo $e->getMessage().'<br>';
				throw new Exception("Tyvärr så är inte ".$otherfiletype." definerad i createwriter ");
				exit();
        }
		
		try
		{
			$objWriter->save($savename);	
		}
		catch( Exception $e ) 
		{
				echo $e->getMessage().'<br>';
				throw new Exception("Tyvärr gick inte ".$savename."  att spara<br>");
				exit();
        }		
		echo $savename.' är sparad.<br>';
		return($justthefilename);
}

Fixformat is a function that copies the templates styles.xml if that will solve this.

tl:dr; styles.xls get corrupted, anyone seen this and have a workaround?
